src/SFA.DAS.EmployerIncentives.Commands/EarningsResilienceCheck/EarningsResilienceApplicationsCheckCommandHandler.cs

```diff
@@ -0,0 +1,29 @@
+﻿using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Queries.EarningsResilienceCheck;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Commands.EarningsResilienceCheck
+{
+    public class EarningsResilienceApplicationsCheckCommandHandler : ICommandHandler<EarningsResilienceApplicationsCheckCommand>
+    {
+        private readonly IIncentiveApplicationDomainRepository _applicationDomainRepository;
+
+        public EarningsResilienceApplicationsCheckCommandHandler(IIncentiveApplicationDomainRepository applicationDomainRepository)
+        {
+            _applicationDomainRepository = applicationDomainRepository;
+        }
+
+        public async Task Handle(EarningsResilienceApplicationsCheckCommand command, CancellationToken cancellationToken = default)
+        {
+            var applications = await _applicationDomainRepository.FindIncentiveApplicationsWithoutEarningsCalculations();
+            foreach (var application in applications)
+            {
+                application.CalculateEarnings();
```

Comment:
Why do this?  Why not just publish the CalculateEarningsCommand ?  There is no need to call a method on the domain object to implement the use case unless the domain object is making some business decision. It doesn't make any sense that the application would calculate earnings.  It's got nothing to do with the application.

So maybe just replace with 

var command = new RefreshEarningsCommand(
                application.AccountId,
                application.IncentiveApplicationId,
                application.AccountLegalEntityId);

            return _commandPublisher.Publish(command)

and have a handler in the ApprenticeshipIncentive domain that re-calculates the calculation or creates the incentive if it doesn't exist using the existing command handlers.

Reply:
See above